src/Sema.zig

```diff
@@ -8992,6 +8992,7 @@ fn zirDeclLiteral(sema: *Sema, block: *Block, inst: Zir.Inst.Index, do_coerce: b
     while (true) switch (ty.zigTypeTag(zcu)) {
```

Comment:
while it definitely allows some pretty silly stuff like your example i think its necessary, imagine something like `fn init() !?*S {`, i dont see any reason this shouldnt work and limiting it would seem arbitrary and potentially pretty complex to get right

Reply:
It might be more fruitful to discuss it in the linked proposal, but as a suggestion I think restricting it to something like the following would make sense:

- `T`
- `?T`
- `*qual T`
- `?*qual T`
- `[*c]qual T`
- `E!T`
- `E!?T`
- `E!*qual T`
- `E!?*qual T`
- `E![*c]qual T`

All but the last 4 are supported as the first parameters in methods/member functions. If we could make the set of types that can participate in decl literals identical to the set that can participate in the method syntax it would make the rules nice and consistent and easy to reason about.

Reply:
This is intentional, and I don't think there's any reason to change it -- many RLS things (such as casting builtins) already strip error unions and optionals from types in a loop (i.e. `const y: ???u32 = @intCast(x)` is fine). Regardless, if you'd like to propose this, please discuss it either in the original proposal or in a new issue.